### PR TITLE
feat!: added basic site structure

### PR DESCRIPTION
**Pull request by trhopkins**
The project skeleton is now here! This will serve as a foundation to build the game off of. We still need tests, CI, assets, and so on, but it's a decent start. See the commit history for a mediocre changelog.

EDIT: here is a basic summary of the things added in this PR:

- Webpack
- Typescript
- .nvmrc
- TailwindCSS
- ESLint
- Jest
- Prettier
- React
- basic site layout
- React Router 6

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        "jsx": "react",
+        "skipLibCheck": true,
+        "outDir": "./dist/",
+        "sourceMap": true,
+        "removeComments": true,
+        "moduleResolution": "node"
+    },
+    "include": [
+        "src/**/*.ts",
+        "src/**/*.tsx",
```
Comment (kyleguarco) [inline tsconfig.json:29]:
```suggestion
        "src/**/*.tsx",
        "test/**/*.ts",
```
ESLint complains that the test modules aren't included. This fixes it.

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ],
+    "exclude": [
+        "node_modules",
+        "dist",
+    ]
+}
```
Comment (kyleguarco) [inline tsconfig.json:35]:
```suggestion
}

```
Include newline?

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+node_modules
+dist
```
Comment (kyleguarco) [inline .eslintignore:2]:
```suggestion
dist

```
Include newline?

Comment (kyleguarco) [reply .eslintignore:2]: Just me being picky

File: tsconfig.json (hunk 2 above)
Comment (kyleguarco) [reply tsconfig.json:36]: Just me being picky

File: .gitignore [hunk 4]
```diff
@@ -1,8 +1,9 @@
-# Node-managed directories
+# Node-managed files
 node_modules/
 dist/
-.vscode/
+package-lock.json
```
[the remark this answers is not in the document]
Comment (kyleguarco) [reply .gitignore:4]: Is there any particular reason why we don't commit the package-lock.json` file?

File: tsconfig.json (hunk 2 above)
Comment (trhopkins) [reply tsconfig.json:36]: > Just me being picky

No code review is too picky! Be as picky as possible!

File: src/print.ts
[the remark this answers is not in the document]
Comment (trhopkins) [reply src/print.ts:1]: Yes, actually. Things in development are fine to keep for testing purposes, but any release versions should be clean for general consumption without excess cruft when possible.

File: .gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (trhopkins) [reply .gitignore:4]: > Is there any particular reason why we don't commit the `package-lock.json` file?

Annoying `git diff` output is one thing. I suppose we can just enforce `npm ci` over `npm i` for environment setup so people don't bump the pinned dependency versions between patches and minor versions. Updating dependencies between major versions that merge to main sounds reasonable, we will just have to be mindful when merging that we should omit actual changes to the `package-lock.json`.

File: test/index.spec.ts
[the remark this answers is not in the document]
Comment (trhopkins) [reply test/index.spec.ts:1]: I'll replace it with a more meaningful set of tests. Since there is no logic aside from routing in the site so far I didn't see a reason to add any tests, but also didn't want to have zero tests altogether. I'll see about using React Testing Library's useNavigate hook with Jest to test the existing code.

**Review by kyleguarco — APPROVED** [review]
No errors when visiting the pages. Production and development builds compile fine.

Both builds still have this error:
```
warn - No utility classes were detected in your source files.
    If this is unexpected, double-check the `content` option in your Tailwind CSS configuration.
warn - https://tailwindcss.com/docs/content-configuration
```

LGTM
